### PR TITLE
feat: Add support for Publisher compression.

### PR DESCRIPTION
**Pull request by Rishabh-V**
Adds support for Publisher compression.

I am creating a document explaining the changes and benefits. I should be able to share it by the end of day today. Thanks.

### CODE REVIEW

**Comment by Rishabh-V** [comment]
I will request review after the document is done and shared. Thanks.

**Review by jskeet — APPROVED** [review]
Seems basically fine to me.

File: apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/PublisherClient.Settings.cs [hunk 1]
```diff
@@ -50,6 +50,20 @@ public sealed class Settings
         /// </summary>
         public TimeSpan? DisposeTimeout { get; set; }
 
+        /// <summary>
+        /// Enables publish message compression. If set to <c>true</c>, messages will be compressed before being sent to the server
+        /// by the <see cref="PublisherClient"/>.
+        /// </summary>
+        public bool EnableCompression { get; set; }
```
[the remark this answers is not in the document]
Comment (Rishabh-V) [reply apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/PublisherClient.Settings.cs:57]: Thanks Jon. Went with `EnableCompression` based on your suggestion in the design doc.

File: apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/PublisherClient.Settings.cs [hunk 2]
```diff
@@ -50,6 +50,20 @@ public sealed class Settings
...
+
+        /// <summary>
+        /// Specifies the threshold for the number of bytes in a message batch before compression is applied.
+        /// This property comes into play only when <see cref="EnableCompression"/> is set to <c>true</c>.
+        /// If the number of bytes in a batch is less than this value, compression will not be applied even
+        /// if <see cref="EnableCompression"/> is <c>true</c>.
```
[the remark this answers is not in the document]
Comment (Rishabh-V) [reply apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/PublisherClient.Settings.cs:63]: Added. Thanks.

**Review by Rishabh-V — COMMENTED** [review]
Incorporated feedback in the new commit. PTAL. Thanks.

**Review by jskeet — APPROVED** [review]
This looks ready to me - was there anything else we're waiting for? (Not sure if Amanda wants to review as well.)

**Comment by Rishabh-V** [comment]
> This looks ready to me - was there anything else we're waiting for? (Not sure if Amanda wants to review as well.)

I was waiting for you to take a look at the review fixes. I will check with Amanda if she wants to have a look as well.

**Comment by Rishabh-V** [comment]
Checked with Amanda. Merging. Thanks.

**Pull request merged**